cppcache/src/Log.cpp

```diff
@@ -502,178 +351,77 @@ LogLevel Log::charsToLevel(const std::string& chars) {
   }
 }
 
-char* Log::formatLogLine(char* buf, LogLevel level) {
-  if (g_pid == 0) {
-    g_pid = boost::this_process::get_id();
-    ACE_OS::uname(&g_uname);
-  }
+std::string Log::formatLogLine(LogLevel level) {
+  std::stringstream msg;
   const size_t MINBUFSIZE = 128;
   auto now = std::chrono::system_clock::now();
   auto secs = std::chrono::system_clock::to_time_t(now);
   auto microseconds = std::chrono::duration_cast<std::chrono::microseconds>(
       now - std::chrono::system_clock::from_time_t(secs));
   auto tm_val = apache::geode::util::chrono::localtime(secs);
-  auto pbuf = buf;
-  pbuf += std::snprintf(pbuf, 15, "[%s ", Log::levelToChars(level));
-  pbuf += std::strftime(pbuf, MINBUFSIZE, "%Y/%m/%d %H:%M:%S", &tm_val);
-  pbuf += std::snprintf(pbuf, 15, ".%06" PRId64 " ",
-                        static_cast<int64_t>(microseconds.count()));
-  pbuf += std::strftime(pbuf, MINBUFSIZE, "%Z ", &tm_val);
+  msg << "[" << Log::levelToChars(level) << " ";
+  char timebuf[MINBUFSIZE];
 
-  std::snprintf(pbuf, 300, "%s:%d %" PRIu64 "] ", g_uname.nodename, g_pid,
-                hacks::aceThreadId(ACE_OS::thr_self()));
+  std::strftime(timebuf, MINBUFSIZE, "%Y/%m/%d %H:%M:%S", &tm_val);
+  msg << timebuf;
 
-  return buf;
-}
+  std::snprintf(timebuf, 15, ".%06" PRId64 " ",
```

Comment:
You should be able to do with with std stream formatting directives and avoid the extra calls to `snprintf`.

Reply:
Done.